src/bindings/jna/libelektra4j/src/main/java/org/libelektra/exception/KDBException.java

```diff
@@ -0,0 +1,74 @@
+package org.libelektra.exception;
+
+import org.libelektra.Key;
+import org.libelektra.exception.model.WarningEntry;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+/**
+ * Custom KDB exception class being used for I/O errors
+ */
+public abstract class KDBException extends java.io.IOException {
+
+	private static final long serialVersionUID = 1L;
+
+	final transient Key errorKey;
+	private Collection<WarningEntry> warnings;
+
+	public KDBException(final Key k) {
+		errorKey = k;
+		Key warningsKey = k.getMeta("warnings");
+		warnings = new ArrayList<>();
+		if (warningsKey.isNull()) {
+			return;
+		}
+		final int nr = warningsKey.getInteger();
+		for (int i = 0; i <= nr; i++) {
+			warnings.add(new WarningEntry(k, nr));
+		}
+	}
+
+	public Key getErrorKey() {
+		return errorKey;
+	}
+
+	public String getErrorCode() {
+		return errorKey.getMeta("error/number").getString();
+	}
+
+	public String getConfigFile() {
+		return errorKey.getMeta("error/configfile").getString();
+	}
+
+	public String getMountpoint() {
+		return errorKey.getMeta("error/mountpoint").getString();
+	}
+
+	public String getDebugInformation() {
+		return String.format("At: %s:%s%n", errorKey.getMeta("error/file").getString(),
+				errorKey.getMeta("error/line").getString());
+	}
+
+	public String getModule() {
+		return errorKey.getMeta("error/module").getString();
+	}
+
+	@Override
+	public String getLocalizedMessage() {
+		return getMessage();
+	}
+
+	@Override
+	public String getMessage() {
```

Comment:
Shouldn't that print the whole message?

Reply:
This is usually just 1-2 sentences and not a big bunch of additional information. It is a bit weird to try to marry C kind of error with java exceptions. I personally would only want to get the Reason message and try to avoid the other verbose and eventually unneccessary output. I can though print everything if you want.

Reply:
It is not about what I want but it is about consistency of error messages. Why should a Java-GUI print different error messages than the Qt-GUI?

Reply:
The QT-GUI also just prints the Reason message and on expansion prints everything. I do not see why they are different

Reply:
I expected that "getReason" will return the reason and "getMessage" will return "everything".

Reply:
getReason is elektra specific while getMessage is java specific. I simply wanted to have both working. But I will change it "everything"

Reply:
All these discussions would not be needed if you had some documentation about what these methods are supposed to do. It is completely unclear what "I simply wanted to have both working" means. What is a working method? One that does not throw exceptions?

Reply:
If someone comes from C (or more specific Elektra) and uses the java binding he would expect "getReason" to be the method with the most "useful" String in it.
If however someone comes from Java he would expect the "getMessage()" method to be the most relevant one.
I wanted to supply both so either of them is satisfied. But you wish for the whole message in the reason part which is also perfectly fine

Reply:
I expected the getMessage() to be the most relevant one. So I wondered why it did not contain the whole error message.

> I wanted to supply both so either of them is satisfied.

Of course it makes sense to have getters for the individual parts but it also makes sense to have a getter for the standardized message format as specified by you in doc/decisions/error_message_format.md

Reply:
You can see an example message now in the README.md

Reply:
Thank you!

But the methods are still not documented.